### ISSUE TITLE
reloading sway crashes firefox (wayland)

### ISSUE THREAD

**Issue by evils** [NixOS/nixpkgs#87962]
**Describe the bug**
reloading (not restarting) sway's config file causes wayland firefox to crash

**To Reproduce**
1. in sway, run wayland native firefox (`firefox-wayland` or `firefox` with `MOZ_ENABLE_WAYLAND=1` environmental variable set)
2. reload sway's configuration (i believe `swaymsg reload` does this though i do it via a keybinding)
3. observe firefox disappearing, stdout contains something like: `Gdk-Message: 02:14:11.949: Lost connection to Wayland compositor.`

**Expected behavior**
sway to reload and firefox staying active just like the other open apps

**Additional context**
found trying to mess with `waybar`, reproduced by @betaboon who's running the same, though likely not related to that

**Notify maintainers**
@andir for firefox (@colemickens maybe interested, if not, sorry for the ping)
@primeos @Synthetica9 for sway

**Metadata**
<details> <summary> nix-info -m </summary>

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.7.0-rc4, NixOS, 20.09.git.954fc73349a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `""`
 - channels(evils): `"nixos-20.09pre225673.8ba41a1e149, nixos-stable-20.03.1866.a7c70f2e10b, nixos-unstable-small-20.09pre225882.e4d710b1c94, nixpkgs-20.09pre225882.e4d710b1c94"`
 - nixpkgs: `/home/evils/.nix-defexpr/channels/nixpkgs`
```
</details>

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: firefox firefox-wayland sway
# a list of nixos modules affected by the problem
module:
```

**Comment by primeos** [comment]
That's strange, on my test PC this does even crash Sway itself... :o
Sometimes I can reload Sway once with neither Sway nor Firefox crashing. 

This is my debug output from Sway:
```
2020-05-16 19:58:33 - [xwayland/xwm.c:1305] unhandled X11 event: MappingNotify (34)
[ This message is repeated 45 times! ]
The XKEYBOARD keymap compiler (xkbcomp) reports:
> Warning:          Unsupported maximum keycode 569, clipping.
>                   X11 cannot support keycodes above 255.
> Warning:          Unsupported high keycode 372 for name <I372> ignored
>                   X11 cannot support keycodes above 255.
>                   This warning only shows for the first high keycode.
Errors from xkbcomp are not fatal to the X server
[ The xkbcomp error is repeated 27 times! ]
2020-05-16 19:58:33 - [main.c:521] Missing a required Wayland interface
Gdk-Message: 19:58:33.647: Error reading events from display: Broken pipe
2020-05-16 19:58:33 - [common/ipc-client.c:88] Unable to receive IPC response
(EE) failed to read Wayland events: Broken pipe
Gdk-Message: 19:58:33.652: Error 104 (Connection reset by peer) dispatching to Wayland display.
Gdk-Message: 19:58:33.647: Error reading events from display: Broken pipe
```

I think the XWayland errors are unfortunately normal, but the following error seems interesting:
```
2020-05-16 19:58:33 - [main.c:521] Missing a required Wayland interface
```

But I think/hope that this is an upstream issue (most likely in Sway or wlroots) and doesn't just affect NixOS (though it might not affect some other distributions due to e.g. different library versions).

**Comment by evils** [comment]
i haven't had sway itself crash
though i just noticed an xwayland app in one of my background workspaces is gone
i may have also had alacritty disappear
so this doesn't seem firefox related, updating issue wording now

**Comment by evils** [comment]
i don't think this is stale

**Comment by evils** [comment]
yea, definitely still happens
now also crashes sway for me

**Comment by primeos** [comment]
> But I think/hope that this is an upstream issue (most likely in Sway or wlroots) and doesn't just affect NixOS (though it might not affect some other distributions due to e.g. different library versions).

I should have noted that more prominently. I think you should report this bug to Sway directly (unless you think it's Nixpkgs specific of course). Be sure to provide all of the required information: https://github.com/swaywm/sway/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by evils** [comment]
this seemed like a big enough issue that if it was due to upstream they'd know about it

a quick search does show they have an open issue that sounds similar
https://github.com/swaywm/sway/issues/6247

**Comment by nazarewk** [comment]
i recently found this https://major.io/2022/05/24/sway-reload-causes-a-firefox-crash/

TLDR; do not use `xkb_numlock` without specific device ID (NOT category),

probably applies to other `xkb_*` options, but `xkb_layout` seems fine
